files/image_config/ntp/ntp-config.sh

```diff
@@ -1,5 +1,37 @@
 #!/bin/bash
 
+ntp_default_file='/etc/default/ntp'
+ntp_temp_file='/tmp/ntp.orig'
+
+reboot_type='cold'
+
+function get_database_reboot_type()
+{
+    SYSTEM_WARM_START=`sonic-db-cli STATE_DB hget "WARM_RESTART_ENABLE_TABLE|system" enable`
+    SYSTEM_FAST_START=`sonic-db-cli STATE_DB get "FAST_REBOOT|system"`
+
+    if [[ x"${SYSTEM_WARM_START}" == x"true" ]]; then
+        reboot_type='warm'
+    elif [[ x"${SYSTEM_FAST_START}" == x"1" ]]; then
+        reboot_type='fast'
+    fi
+}
+
+function modify_ntp_default
+{
+    cp ${ntp_default_file} ${ntp_temp_file}
+    sed -e "$1" ${ntp_temp_file} >${ntp_default_file}
```

Comment:
Would `sed -i` work better?

Reply:
sed -i is probably safe for this case. But as Joe said, we discovered quite a few file contents corruptions related to using sed -i. Currently, we are avoiding sed -i as much as we could.